core/msg.c

```diff
@@ -322,6 +322,11 @@ int msg_reply_int(msg_t *m, msg_t *reply)
 {
     thread_t *target = thread_get_unchecked(m->sender_pid);
 
+    if (target == NULL) {
+        DEBUG("msg_reply_int(): target thread %d does not exist\n", m->sender_pid);
+        return -1;
+    }
+
```

Comment:
Hmm, this looks to me like it can only trigger when in violation with the API. E.g. `msg_reply_int()` must make sure that the thread to reply to does exists (e.g. in an app where no thread ever exits, this is trivial).

Adding an `assert()` here with a comment that `msg_reply_int()` can only be used to reply to existing threads could help debugging, though.

@kaspar030 Could we deprecate all the `msg*_ist()` functions? IMO apps should migrate to `msg_post()` for IRQ to thread communication anyway.